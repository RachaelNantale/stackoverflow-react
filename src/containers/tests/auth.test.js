import React from 'react';
import { shallow } from 'enzyme';
import AuthPageConnected, { mapDispatchToProps, AuthPage } from '../auth';

import ACTION_TYPE from '../../actions/actionTypes';

describe('<AuthPage />', () => {
  let AuthPageComponent;
  let wrapper;
  beforeEach(() => {
    AuthPageComponent = shallow(<AuthPageConnected />);
  });

  it('should render the component', () => {
    expect(AuthPageComponent).toMatchSnapshot();
  });

  it('should dispatch a method to  get   signup user input', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).getUserInputs({});
    expect(dispatch.mock.calls[0][0]).toEqual({ payload: {}, type: ACTION_TYPE.GET_USER_INPUT });
    mapDispatchToProps(dispatch).signUpuser({});
  });

  it('should dispatch a method to get  login user input', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).getUserInputs({});
    expect(dispatch.mock.calls[0][0]).toEqual({ payload: {}, type: ACTION_TYPE.GET_USER_INPUT });
    mapDispatchToProps(dispatch).loginUser({});
  });

  it('should call handle input user method', () => {
    const getUserInputs = jest.fn();
    wrapper = shallow(
      <AuthPage
        signUpuser={jest.fn()}
        loginUser={jest.fn()}
        freshUser={{}}
        getUserInputs={getUserInputs}
      />,
    );
    wrapper.instance().handleUpdateFields({ target: { name: 'username', value: 'rachael' } });
    expect(getUserInputs).toHaveBeenCalled();
  });
});

describe('handle signup', () => {
  const wrapper = shallow(<AuthPage />);
  wrapper.setProps({ signUpuser: jest.fn() });
  const fakeEventReturn = { target: { id: 1, value: 'some val' } };
  const fakeEvent = { preventDefault: () => fakeEventReturn };
  const spy = jest.spyOn(wrapper.instance(), 'handleSignUp');
  wrapper.instance().handleSignUp(fakeEvent);
  expect(spy).toHaveBeenCalled();
});

describe('handle login', () => {
  const wrapper = shallow(<AuthPage />);
  wrapper.setProps({ loginUser: jest.fn() });
  const fakeEventReturn = { target: { id: 1, value: 'some val' } };
  const fakeEvent = { preventDefault: () => fakeEventReturn };
  const spy = jest.spyOn(wrapper.instance(), 'handleLogin');
  wrapper.instance().handleLogin(fakeEvent);
  expect(spy).toHaveBeenCalled();
});
