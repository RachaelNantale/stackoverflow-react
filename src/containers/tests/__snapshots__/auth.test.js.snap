// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthPage /> should render the component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
